app/Models/ResourceItem.php

```
@@ -61,53 +67,73 @@ public function scopeFilterByResourceType($query, $resourceType)
             ->unique();
-        $affectedResources = self::whereHas('task.phase', function ($query) use ($projectId) {
-            $query->where('project_id', $projectId);
-        })
-            ->where('resource_type', $this->resource_type)
-            ->where('unit', $this->unit)
-            ->where('description', $this->description)
-            ->where('id', '!=', $this->id)
+        $affectedResources = self::matchingResources($projectId, $this)
             ->update([
                 'unit_cost'  => $this->unit_cost,
                 'total_cost' => DB::raw('quantity * ' . (float) $this->unit_cost),
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\DB' in method 'syncUnitCostAcrossProjectResources'. (undefined)

(StaticAccess)